pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -180,6 +185,32 @@ type ResourceSelector struct {
 	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
 }
 
+// FilterResourceSelector the resources will be filtered.
+type FilterResourceSelector struct {
```

Comment:
Should there also be a mandatory requirement for resources in FilterResourceSelector to also be in ResourceSelectors?

Reply:
Why do you think that? I was thinking about it too. But don't have a clue yet.

Reply:
> Should there also be a mandatory requirement for resources in FilterResourceSelector to also be in ResourceSelectors?

Sorry, I didn't get it, can you give a simple example?

Reply:
@chaunceyjiang Can you explain your opinion a little more? 
Besides, I've updated the API, could you please see if there are any problems with the latest version?

Reply:
例如：

```
resourceSelectors:

- apiVersion: aaa/v1
   kind: cluster
- apiVersion: bbb/v1
   kind: cluster             # Newly added kind

excludedResources:
- kind: cluster
```

Will this newly added 'kind' resource propagate to the member clusters?

Reply:
I think the answer is `no`, as the newly added resource already been `excluded`.

Reply:
@chaunceyjiang 
> Will this newly added 'kind' resource propagate to the member clusters?

No, because `aaa/v1/cluster` and `bbb/v1/cluster` are all hit by `excludedResources` condition `kind: cluster`, so neither `aaa/v1/cluster` nor `bbb/v1/cluster` be propagate to the member clusters.

Reply:
```
resourceSelectors:

- apiVersion: aaa/v1
   kind: cluster
- apiVersion: aaa/v1
   kind: cluster
- apiVersion: ccc/v1
   kind: cluster             # Newly added kind

excludedResources:
- kind: cluster
  name: bar
```

> so neither aaa/v1/cluster nor bbb/v1/cluster be propagate to the member clusters.


Yes, this is exactly what I was worried about. I added a new GVK, but it surprisingly couldn't be propagated to the member clusters. Apparently, when filling out excludedResources, there was no intention to filter this resource.


So, I believe that the benefit of setting `apiVersion` and `kind` optional to reduce user costs is really minimal.

Reply:
> So, I believe that the benefit of setting apiVersion and kind optional to reduce user costs is really minimal.

Thank you for your suggest, but I think this has nothing to do with it setting `apiVersion` and `kind` optional, this is because you did not follow the rules when writing the policy.

Even if we set `apiVersion` and `kind` required, consider the following example:

```yaml
resourceSelectors:
- apiVersion: aaa/v1
   kind: cluster
   namespace: a
- apiVersion: aaa/v1
   kind: cluster
   namespace: b       # newly added namespace
excludedResources:
- apiVersion: aaa/v1
   kind: cluster
```

now `apiversion` and `kind` is required, then you add a `nameapce: b` resourceSelectors, but it couldn't be propagated to the member clusters, this is essentially the same principle as the example you provided above.

I think it essentially not surprising, but you did not follow the rules when updating the policy.

Reply:
In your case, if you truly want to exclude `cluster` resources under the specified `apiversion`, you should write it fully in the `excludedResources` item. Correspondingly, if you don't write it fully, we can assume that you are trying to exclude `cluster` resources under any `apiversion`, and that the modifier himself should pay attention to the `excludedResources` field when updating `resourceSelectors`.